### PR TITLE
Infinite Gun Culling: Gun Kit Edition

### PR DESCRIPTION
**Pull request by necromanceranne**
## About The Pull Request

Every prototype firearm in the game now utilizes crafting to construct it, usually sacrificing a standard version of that firearm in order to produce the new one. The guns are made using gun part kits you print from the security protolathe (or buy, in the case of hellfire lasers).

**The weapons that utilize Energy Guns in their crafting include:**
- Advanced Energy Guns (duh)
- Temperature Guns
- Particle Acceleration Rifles

**The weapons that utilize Laser Guns in their crafting include:**
- X-ray Laser Guns (duh again)
- Hellfire Laser Guns (this is a given)
- Decloner
- Ion Carbine

**Some weapons are outliers and use different parts**
- Particle Acceleration Rifles use a flux anomaly and a gravity anomaly as part of their construction.
- Tesla Revolvers use flux anomalies as part of their construction and need no other gun. You build the gun around the anomaly!
- Decloners require 40 units of mutagen (their old protolathe recipe) and 30 units of baldium (ayy lmao).
- Energy Crossbows require a Proto-Kinetic Accelerator and 15 units of radium to construct. 

In addition:

Hellfire Laser Guns now come as gun part kits and are significantly cheaper as goodies.

Energy Crossbows (the big ones) are bulky. They are called large for a reason.

Every weapon that was printed from the lathe now actually spawns with a firing pin when constructed.

## Why It's Good For The Game

@Timberpoes I've somewhat disappointed you in NOT making them use anomalies but...

I didn't think it entirely fair to lock x-rays behind specifically the limited station supply of anomalies, but I did think they should be more limited. Now, they're limited by station supply and cargo credits. All the more reason for security to contribute to the station funds.

**Real Talk**
Somewhere along the lines we decided that printing out Saber SMGs from the science protolathe was a bad idea, but yet here we are with a ton of some quite powerful weapons freely printable from our current lathes. Then we moved our guns to the security protolathe and the results were pretty much roughly equivalent in my opinion, a gun printer is a gun printer whether it's in science or security, and a bunch of easily produced guns is easily produced guns for the whole crew. If we want to limit weapon access,  we can't have these weapons stay in the state they're in. Not if we want cargo to represent an important part in gun distribution.

On the other side of this coin, I am fairly sure I heard people calling for the removal of beam rifles all the way back when I started playing the game in mid 2018. They rendered blob a cakewalk, and you had plenty of firing pins to just print off half a dozen and decimate any blob threat on the station regardless of where they are or how well they're doing. This change doesn't stop them from being good against blobs, but it should help justify giving beam rifles a broader niche than JUST blob killing, since I'm of the opinion they were deliberately handicapped explicitly because of their mass producibility. It is out of scope for this PR, but I hope this will give some justification for making the beam rifle a weapon worth using in a separate PR.

## Changelog
:cl:
add: Guns are no longer directly printable from the protolathe.
add: Gun kits to craft the various prototype firearms once printable from the security protolathe.
balance: Large energy crossbows now are bulky, like most other printed weapons.
balance: Crafted weapons now come with a firing pin.
/:cl:

### CODE REVIEW

**Comment by necromanceranne** [comment]
gun kit

**Comment by Cartographer-D** [comment]
Making it so every gun starts with an electronic pin is a pretty big balance change.

Currently, you have a limited number of non-mindshield firing pins, which are available in a separate room (Secure Gear Storage).

If the advanced guns are researched, you also have the mindshield pins, but those only work on mindshielded people, and there are a limited number of extra mindshields.

**So what this means, effectively, is you can mass print weapons to hand out to the entire crew with no recompense on modes like nukies, wizard, revolution, cult, etc.**

Everything else is great imo, but I have serious problems with the pins.

**Comment by necromanceranne** [comment]
There is literally zero net loss or gain in weapons with these changes, so if you had the base weapons to begin with, you are effectively taking them and making them into a new weapon. 

It wouldn't matter if the kits produced guns with pins or not. They used weapons with pins to begin with. And if you mulched you gun without a pin to put into your new gun, that's...just kind of a feels bad, you know?

**Comment by Cartographer-D** [comment]
I don't understand what you're saying. Do the recipes require the gun to have the pin installed? If so, make the recipe either have a pin or not have a pin based on if the base gun had a pin.

You're wrong to say that having the pins at the start isn't a massive buff. Previously, you had to use the limited supply of pins if you wanted to distribute weapons from the lathe. Now, every weapon that can be printed from the lathe can be mass distributed. Not good.

**Comment by Cartographer-D** [comment]
Are you saying that using a gun kit installs a pin here?
image

**Comment by Tlaltecuhtli** [comment]
why not attackby on gun with mod?

File: code/datums/components/crafting/recipes.dm [hunk 1]
```diff
@@ -142,6 +142,142 @@
...
+
+/datum/crafting_recipe/decloner/New()
+	..()
+	blacklist += subtypesof(/obj/item/gun/energy/laser)
+
+/datum/crafting_recipe/teslarevolver
+	name = "Tesla Revolver"
+	tools = list(TOOL_SCREWDRIVER, TOOL_MULTITOOL)
+	result = /obj/item/gun/energy/tesla_revolver
+	reqs = list(/obj/item/assembly/signaler/anomaly/flux = 1,
```
Comment (Tlaltecuhtli) [inline code/datums/components/crafting/recipes.dm:274]: does this garbage of gun really need an anomaly?

**Comment by necromanceranne** [comment]
> why not attackby on gun with mod?

I've been asked to not use slapcraft so _shrug_.

> You're wrong to say that having the pins at the start isn't a massive buff. Previously, you had to use the limited supply of pins if you wanted to distribute weapons from the lathe. Now, every weapon that can be printed from the lathe can be mass distributed. Not good.

You seem to be confused as to how this process is working, so lemme re-articulate this as best I can.

You use the kits you print out of the protolathe as an ingrediant in the recipe for making the weapon you want from the kit. This includes previously printed directly from the protolathe.

To do this, many of the recipes use either a laser gun or an energy gun. Yes, the end result might grant said weapon a firing pin. But the weapons that you use to make those weapon will, upon initialization, have a firing pin added to those weapons.

Every single energy gun or laser gun that appears in the game roundstart will have a firing pin. If somewhere in this process said firing pin is removed, I literally don't fucking _care_. What does it matter? That's an outlier situation not worth worrying about because few people deliberately go out of their way to unpin the station weapons to leave them unpinned on purpose. There are plenty more lasers where that came from, and under most any other circumstance, if you managed to acquire that energy gun or laser gun, chances are it has a firing pin already, and thus is an entirely workable weapon as is for your shooting needs. They come in cargo crates pre-pinned. They spawn in the armory pre-pinned. I picked them explicitly because they are so ubiquitous and reasonable to acquire.

Does that help?

File: code/datums/components/crafting/recipes.dm (hunk 1 above)
Comment (necromanceranne) [reply code/datums/components/crafting/recipes.dm:274]: Did you read the open post? I said adding anomalies to the recipes gives reason to make these weapons better. They suck right now because they've been hugboxed to shit.

**Comment by Cartographer-D** [comment]
Yup! We're good!

File: code/datums/components/crafting/recipes.dm (hunk 1 above)
Comment (Fikou) [reply code/datums/components/crafting/recipes.dm:274]: but this pr doesnt make them better, and i imagine nobody will care enough to buff it

**Comment by ATH1909** [comment]
"Particle Acceleration Rifles use a flux anomaly and a gravity anomaly as part of their construction."
Blob winrate skyrockets to 200%.

Forgive me if this sounds dumb, but, uh, how is the crew currently expected to deal with a blob WITHOUT spamming beam rifles?

**Comment by ShibaInuLord** [comment]
> 
> 
> "Particle Acceleration Rifles use a flux anomaly and a gravity anomaly as part of their construction."
> Blob winrate skyrockets to 200%.
> 
> Forgive me if this sounds dumb, but, uh, how is the crew currently expected to deal with a blob WITHOUT spamming beam rifles?

literally everything else

**Comment by necromanceranne** [comment]
Lasers are usually pretty good. I've seen blobs lose to crew without beam rifles. I genuinely do not believe you need to have them. But when they pop up, beam rifles are typically a death knell for the blob.

If they only thing keeping blobs from autowinning is an autoloss in the form of a beam rifle, that's...entirely unrelated to this PR and is in fact a good time to have a hard look at the balance of blob.

File: code/datums/components/crafting/recipes.dm (hunk 1 above)
Comment (necromanceranne) [reply code/datums/components/crafting/recipes.dm:274]: I don't want to do too much in one PR. If I overbloat this one with a bunch of unrelated balance changes, it'll only slow things down. One step at a time.

**Comment by Tlaltecuhtli** [comment]
can you set the craft time to something normal like 5s instead of 20 whole second standing still?

**Comment by SkeletalElite** [comment]
Too bad someone has to actually research the tech for these guns first before someone could even think of getting a new gun and that happens almost never.

**Comment by CRITAWAKETS** [comment]
> * Particle Acceleration Rifles use a flux anomaly and a gravity anomaly as part of their construction.

the worst weapon in the game which can literally only kill blobs who suck and nothing else now costs more than the phazon, sounds good to me

**Comment by necromanceranne** [comment]
> > * Particle Acceleration Rifles use a flux anomaly and a gravity anomaly as part of their construction.
> 
> the worst weapon in the game which can literally only kill blobs who suck and nothing else now costs more than the phazon, sounds good to me

I'm not going to repeat myself a whole ton on this, as I specifically mentioned beam rifles and why they are being as a direct result of their mass production issue, but I'm not touching specific gun balance in a general gun restriction pr. I totally acknowledge that the weapon is terrible beyond blob. I also don't think make any changes to that in this PR is a wise decision.

**Comment by D4C-420** [comment]
Correct me if I’m wrong but are firing pins not the way guns are limited? By stopping sec from printing off a mind shielded firing pin you can prevent this infinite gun issue.  Cargo can already buy more firing pins. Rather than forcing sec to waste time crafting new guns (and preventing the distribution of guns during say nuke ops by limiting their numbers severely ) behind an already time limited gate just remove some of the round start firing pins (leave like 1-2 in case of the need for replacement) and take out test range (because an Emag just makes them regular firing pins and sec will abuse that) and mind shielded firing pins. That way they still have to bother cargo for firing pins to make actual use of the guns. 

Tldr: just remove printable firing pins if you don’t want sec from printing guns for themselves and remove a few roundstart firing pins from the armory to prevent mass distribution of guns to the crew.

**Comment by necromanceranne** [comment]
> Correct me if I’m wrong but are firing pins not the way guns are limited? By stopping sec from printing off a mind shielded firing pin you can prevent this infinite gun issue.

I don't think pins are doing quite enough. I was saying to people in coding-general how often mindshield pins don't really do their job very well in being a way to prevent people from getting access too these weapons, since any roundtype that relies on mindshields often end far too quickly for mindshield pins to see use. They're late, late tech, and a lot of these guns are as well. Cult and Rev rounds end very quickly, or stagnate hard, preventing any research being done in the first place. But once you have them, the mindshield pins allow security to never bother with cargo.

I personally would actually like to see mindshield firing pins be available earlier than later rather than remove them outright or use them as the basis of gun availability balance. Allow security the chance to get their weapons secured in time for opponents who might want to grab them from their corpse, and not force a premium item into being a balance point for guns as a whole. Something like mindshield pins should be a QoL benefit for security and an effective sidegrade at most to the standard electronic pin.

>Cargo can already buy more firing pins. Rather than forcing sec to waste time crafting new guns (and preventing the distribution of guns during say nuke ops by limiting their numbers severely ) behind an already time limited gate just remove some of the round start firing pins (leave like 1-2 in case of the need for replacement) and take out test range (because an Emag just makes them regular firing pins and sec will abuse that) and mind shielded firing pins. That way they still have to bother cargo for firing pins to make actual use of the guns.
> 
> Tldr: just remove printable firing pins if you don’t want sec from printing guns for themselves and remove a few roundstart firing pins from the armory to prevent mass distribution of guns to the crew.

This is an entirely arbitrary limitation achieving the same results but with less player signposting as to what the next step in the procedure is to getting an operational new weapon, while attached to a fairly redundant system to begin with. I'll leave cargo gun bloating issues to cargo balance, and remove security gun bloat as an additional factor. It seems like a win-win for weapon control to make cargo the sole domain for gun acquisition, since it is less factors to have to worry about.

Moreso, cargo gets a significant discount if they're ordering electronic firing pins compared to ordering guns wholesale. If security is printing the guns, you're paying maybe a 10th of the price of a gun out of cargo per pin in actual credits, while the only limit for security gun printing is material costs. If cargo is flushed enough to be ordering pins en-masse, it's a safe assumption security has the materials to print off guns too.

This method at least keeps it 1:1 since a gun is consumed in the process of making the experimental weapons, and so the security weapons don't jeopardize cargo gun credit values in any way. The alternative to this is charging the full price of a gun per firing pin, which is I guess an option but a shitty option only because none of the pins in a round are actually recoverable once used, as well as the spaghetti weirdness that occurs every so often as a result of gun operation being reliant on the contents of those guns. Like when material datum was first introduced, and guns were being bricked when blown up from their pins + cells being deleted, which couldn't be reversed. Oddities like that wouldn't happen if it wasn't using actual objects for the same functionality.

(I've also never seen the experimental weapons make much of an impact in nukie rounds. they are all too expensive per node most of the time and you're only reasonably able to rush toward WT-550 ammo in a war round a lot of the time, unless you're using bombs)

**Comment by D4C-420** [comment]
@necromanceranne thanks for the explanation on why you did this like ya did. Was mainly wondering why not use the already in place system for limiting the amount of guns. Was not aware of the code issue I’m not on the discord or even understand a tenth of the code.   

On a side note you would be a dick for doing it but it’s is possible to rush up to advance weapons with the aid of strange objects, toxins and Xenobiology. Wether or not mining has gotten enough ore to print shit is a separate issue.

**Comment by Pugie** [comment]
Yep, this PR sounds good to me, infinitely printed guns at techlathe always seemed dumb to me.

**Comment by Tlaltecuhtli** [comment]
can you mappedit revert the seclathe in armory change from #55886 ? as the guns are no longer printable without having a gun in your hand already so there isnt a need to lock modkits in armory, solving the OP's issue and making  rest of the printable items available without warden supervision

File: code/datums/components/crafting/recipes.dm [hunk 2]
```diff
@@ -142,6 +142,142 @@
 	category = CAT_WEAPONRY
 	subcategory = CAT_WEAPON
 
+/datum/crafting_recipe/advancedegun
+	name = "Advanced Energy Gun"
+	tools = list(TOOL_SCREWDRIVER, TOOL_MULTITOOL)
+	result = /obj/item/gun/energy/e_gun/nuclear
+	reqs = list(/obj/item/gun/energy/e_gun = 1,
```
Comment (Qustinnus) [inline code/datums/components/crafting/recipes.dm:149]: The fact that this creates guns with a firing pin when there could be no firing pin in the original gun is a bit weird. Can you add an override to check_requirements that checks if a pin is installed?

**Comment by necromanceranne** [comment]
That PR only just got merged, why would I add a revert in this one?

**Review by ghost — CHANGES_REQUESTED** [review]
one last thing; I think there might be some mappable subtypes which by default have a pin, seeing as this PR removes the lack of a pin in these weapons, maybe the subtypes could also be removed?

**Review by ghost — CHANGES_REQUESTED** [review]
one last thing; I think there might be some mappable subtypes which by default have a pin, seeing as this PR removes the lack of a pin in these weapons, maybe the subtypes could also be removed?

**Comment by necromanceranne** [comment]
> one last thing; I think there might be some mappable subtypes which by default have a pin, seeing as this PR removes the lack of a pin in these weapons, maybe the subtypes could also be removed?

Yeah I was thinking that too. I'll go through and have a look for any duplicates.

File: code/datums/components/crafting/recipes.dm (hunk 2 above)
Comment (necromanceranne) [reply code/datums/components/crafting/recipes.dm:149]: Honestly, I wonder if it'd be more functionally better to find a method to have the gun spawned not have a pin if the base gun didn't...

**Comment by Tlaltecuhtli** [comment]
> That PR only just got merged, why would I add a revert in this one?
cause the point was to have funny printable guns more restricted, but this does it already

**Comment by necromanceranne** [comment]
> > That PR only just got merged, why would I add a revert in this one?
> cause the point was to have funny printable guns more restricted, but this does it already

Ain't my business. Ask a maptainer if this PR gets merged to restore access to the lathe.

File: code/datums/components/crafting/recipes.dm [hunk 3]
```diff
@@ -142,6 +142,142 @@
...
+	time = 200
+	category = CAT_WEAPONRY
+	subcategory = CAT_WEAPON
+
+/datum/crafting_recipe/tempgun/New()
+	..()
+	blacklist += subtypesof(/obj/item/gun/energy/e_gun)
+
+/datum/crafting_recipe/beam_rifle
+	name = "Particule Acceleration Rifle"
```
Comment (Fikou) [inline code/datums/components/crafting/recipes.dm:176]: particule

File: code/datums/components/crafting/recipes.dm [hunk 4]
```diff
@@ -142,6 +142,142 @@
 	category = CAT_WEAPONRY
 	subcategory = CAT_WEAPON
 
+/datum/crafting_recipe/advancedegun
+	name = "Advanced Energy Gun"
+	tools = list(TOOL_SCREWDRIVER, TOOL_MULTITOOL)
```
Comment (Fikou) [inline code/datums/components/crafting/recipes.dm:147]: multitool see,s weird, its a tool security doesnt have by default and we shouldnt expect them to grab one, why not crowbar/wrench? they are in the wardens office and easier to get

File: code/modules/projectiles/guns/energy/special.dm [hunk 5]
```diff
@@ -112,11 +110,10 @@
 	name = "energy crossbow"
 	desc = "A reverse engineered weapon using syndicate technology."
 	icon_state = "crossbowlarge"
-	w_class = WEIGHT_CLASS_NORMAL
+	w_class = WEIGHT_CLASS_BULKY
```
Comment (Fikou) [inline code/modules/projectiles/guns/energy/special.dm:113]: does this fit on any slot or do you have to carry it on your belt

Comment (necromanceranne) [reply code/modules/projectiles/guns/energy/special.dm:113]: It should fit into armor vests

File: code/datums/components/crafting/recipes.dm (hunk 4 above)
Comment (necromanceranne) [reply code/datums/components/crafting/recipes.dm:147]: I swear security's toolbox comes with a multitool instead of a analyzer on at least metastation. And I could swear the other stations have one too either in the same toolbox or just lying out. Either in the warden's office or in the brig itself. Though I could be wrong.

File: code/datums/components/crafting/recipes.dm (hunk 3 above)
Comment (necromanceranne) [reply code/datums/components/crafting/recipes.dm:176]: hate

File: code/datums/components/crafting/recipes.dm (hunk 2 above)
Comment (Qustinnus) [reply code/datums/components/crafting/recipes.dm:149]: it would be but thatll be harder.

File: code/datums/components/crafting/recipes.dm (hunk 4 above)
Comment (Fikou) [reply code/datums/components/crafting/recipes.dm:147]: i dont think so?

File: code/datums/components/crafting/recipes.dm (hunk 1 above)
Comment (wesoda25) [reply code/datums/components/crafting/recipes.dm:274]: Ill buff them eventually if no one else does

File: code/datums/components/crafting/recipes.dm (hunk 4 above)
Comment (Fikou) [reply code/datums/components/crafting/recipes.dm:147]: mappers dont have any ability to change what gets created in storage items and i dont think thres a sec subtype of any toolbox

**Comment by wesoda25** [comment]
This is good except particle acceleration rifles should not call for two anomalies. AFAIK that’s currently unprecedented, and beam rifles aren’t powerful enough to justify such a thing.

**Comment by Cartographer-D** [comment]
> 
> 
> > That PR only just got merged, why would I add a revert in this one?
> > cause the point was to have funny printable guns more restricted, but this does it already

If you paid attention, you'd have realized that the PR moved the lathe back to the offices before it was merged.

Stop trying to get other people to revert work that was literally just done. That's a scumbag move.

**Comment by necromanceranne** [comment]
> This is good except particle acceleration rifles should not call for two anomalies. AFAIK that’s currently unprecedented, and beam rifles aren’t powerful enough to justify such a thing.

I have spoken already that any change to beam rifles is out of scope for the pr. This should serve as justification for improving them. It is unprecedented, but that's what lends credit to making them stronger possibly.

**Review by Twaticus — APPROVED** [review]

**Comment by ATH1909** [comment]
> > This is good except particle acceleration rifles should not call for two anomalies. AFAIK that’s currently unprecedented, and beam rifles aren’t powerful enough to justify such a thing.
> 
> I have spoken already that any change to beam rifles is out of scope for the pr. This should serve as justification for improving them. It is unprecedented, but that's what lends credit to making them stronger possibly.

Can't you just... put the cost increase in the same PR that buffs the beam rifle?

Make the beam rifle cost ONE of either of those two cores to build if you're worried about bullying blobs. Make it cost TWO cores in whatever future PR you make that buffs beam rifles. There's no need for it to undergo a period of being insanely overpriced for what it does.

**Comment by ATH1909** [comment]
Also, uh, are you sure that it's a good idea to even HAVE a gun in the game that's of a power level appropriate for the expenditure of two specific kinds of anomaly cores? Like, unless you turn the beam rifle into a goddamn taser or a no charge up time, rechargeable missile launcher or something, it's not gonna be worth the time and effort (and wrangled cats) you'd have to go through to obtain it. I really don't think that that much power should be in the hands of a single person/gun.

**Comment by Qustinnus** [comment]
cool story ath

**Comment by wesoda25** [comment]
> I have spoken already that any change to beam rifles is out of scope for the pr. This should serve as justification for improving them. It is unprecedented, but that's what lends credit to making them stronger possibly.

??? Or instead of nerfing them to force a buff, you could just make them use one anomaly so that they remain balanced.

**Comment by necromanceranne** [comment]
You'll get used to it.

**Comment by ATH1909** [comment]
> You'll get used to it. I've made my point at least three times now. I'm not arguing it further.

if we'll get used to it, how is it supposed to passive aggressively nudge people into maybe deciding to buff the beam rifle?

File: code/datums/components/crafting/recipes.dm (hunk 4 above)
Comment (necromanceranne) [reply code/datums/components/crafting/recipes.dm:147]: I'm gonna drop this down to just screwdriver and wirecutters for now, just because I think you're right about the multitools being a dubious feature of security.

**Comment by Timberpoes** [comment]
Okay.

It'll make the armory more important, sec may now need to protect it better or alternatively interface with cargo for weapons out of necessity instead of convenience.

I want to see how this shakes up the meta, Also allows for more streamlined changes to weapons - They can be nerfed with additional crafting requirements and buffed with additional power which can be locked behind additional crafting requirements.

**Review by Timberpoes — APPROVED** [review]

**Comment by adamkad1** [comment]
xray guns are kinda eh, particle rifles needing a core means blobs are going to win all the time now
